server/ladder_service/ladder_service.py

```diff
@@ -523,7 +534,25 @@ def get_displayed_rating(player: Player) -> float:
             pool = queue.get_map_pool_for_rating(rating)
             if not pool:
                 raise RuntimeError(f"No map pool available for rating {rating}!")
-            game_map = pool.choose_map(played_map_ids)
+            
+            pool, _, _, veto_tokens_per_player, max_tokens_per_map, minimum_maps_after_veto = queue.map_pools[pool.id]
+
+            vetoesMap = defaultdict(int)
+            tokensTotalPerPlayer = defaultdict(int)
+
+            for (id, map) in pool.maps.items():
+                for player in all_players:
+                    vetoesMap[map.map_pool_map_version_id] += player.vetoes.get(map.map_pool_map_version_id, 0)
+                    tokensTotalPerPlayer[player.id] += player.vetoes.get(map.map_pool_map_version_id, 0)
+
+            for player in all_players:
+                if (tokensTotalPerPlayer[player.id] > veto_tokens_per_player):
+                    raise RuntimeError(f"Player {player.id} has too many vetoes!")
```

Comment:
We should not do validation of the number of tokens per player when starting the game. This validation and error raising should happen when the server receives the set_vetoes message. This is because as it is now a player could send the server more than the max amount of vetoes and then cause a game to never be able to launch.